L1Trigger/TrackTrigger/plugins/TTStubBuilder.cc

```diff
@@ -0,0 +1,327 @@
+/*!  \brief   Implementation of methods of TTClusterBuilder.h
+ *  \details Here, in the source file, the methods which do depend
+ *           on the specific type <T> that can fit the template.
+ *
+ * \author Andrew W. Rose
+ * \author Nicola Pozzobon
+ * \author Ivan Reid
+ * \date 2013, Jul 18
+ *
+ */
+
+#include "L1Trigger/TrackTrigger/plugins/TTStubBuilder.h"
+
+/// Implement the producer
+template< >
+void TTStubBuilder< Ref_Phase2TrackerDigi_ >::produce( edm::Event& iEvent, const edm::EventSetup& iSetup )
+{
+  //Retrieve tracker topology from geometry                                                                                                              
+  edm::ESHandle<TrackerTopology> tTopoHandle;
+  iSetup.get<TrackerTopologyRcd>().get(tTopoHandle);
+  const TrackerTopology* const tTopo = tTopoHandle.product();
+  edm::ESHandle< TrackerGeometry > tGeomHandle;
+  iSetup.get< TrackerDigiGeometryRecord >().get( tGeomHandle );
+  const TrackerGeometry* const theTrackerGeom = tGeomHandle.product();
+	
+  /// Prepare output
+  std::auto_ptr< edmNew::DetSetVector< TTCluster< Ref_Phase2TrackerDigi_ > > > TTClusterDSVForOutput( new edmNew::DetSetVector< TTCluster< Ref_Phase2TrackerDigi_ > > );
```

Comment:
One other thing, `std::auto_ptr` is decremented in the C++ standard so we should be switching to `std::unique_ptr` (which is what `std::make_unique` returns).